dali/kernels/slice/slice_gpu.cuh

```diff
@@ -237,14 +240,22 @@ class SliceGPU {
 
     std::vector<int64_t> sample_sizes;
     sample_sizes.reserve(slice_args.size());
+    int64_t total_volume = 0;
     for (auto &args : slice_args) {
       sample_sizes.push_back(volume(args.shape));
+      total_volume += volume(args.shape);
+    }
+
+    auto minBlocks = 4 * GetSMCount();
+    blockSize = kMaxBlockSize;
+    while (total_volume / blockSize < minBlocks && blockSize > kMinBlockSize) {
+      blockSize /= 2;
     }
 
     block_count_ = 0;
     for (auto sample_size : sample_sizes) {
       block_count_ += std::ceil(
-        sample_size / static_cast<float>(kBlockSize));
+        sample_size / static_cast<float>(blockSize));
```

Comment:
This is prone to rounding errors. Use `div_ceil` instead.

Reply:
Changed it to `div_ceil`, thank you